src/GUI/ComboBoxColorPicker.java

```diff
@@ -1,42 +1,43 @@
+package GUI;
+
 import javax.swing.*;
 import java.awt.*;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
 
 class ComboBoxColorPicker extends JPanel {
     private final Color[] graphColors = {Color.RED, Color.BLUE, Color.YELLOW, Color.GREEN, Color.GRAY};
-    private final JComboBox<Color> colorPicker = new JComboBox<>(graphColors);   // Combo Box to pick the Color of the Function
+    private final JComboBox<Color> colorPicker = new JComboBox<>(graphColors);
+    public Color currentColor;
```

Comment:
avoid state where possible. Rather, let's have a function called 'getCurrentColor' (or similar) and just return

`(Color) colorPicker.getSelectedItem()`

from there. 'currentColor' is not needed as far as I can see.

Reply:
Done